src/java/htsjdk/samtools/BamIndexValidator.java

```diff
@@ -84,4 +88,64 @@ public static int exhaustivelyTestIndex(final SamReader reader) { // throws Exce
         return 0;
     }
 
+    /**
+     * A less time-consuming index validation that only looks at the first and last references in the index
+     * and the first and last chunks in each of those
+     *
+     * @param reader
+     * @return # of chunks examined, or 0 if there is no browseable index for the reader
+     */
+    public static int lessExhaustivelyTestIndex(final SamReader reader) {
+        // look at all chunk offsets in a linear index to make sure they are valid
+
+        if (reader.indexing().hasBrowseableIndex()) {
+
+            // content is from an existing bai file
+            final CachingBAMFileIndex existingIndex = (CachingBAMFileIndex) reader.indexing().getBrowseableIndex();
+            final int numRefs = existingIndex.getNumberOfReferences();
+
+            int chunkCount = 0;
+            int indexCount = 0;
+            for (int i = 0; i < numRefs; i++) {
+
+                final BAMIndexContent content = existingIndex.getQueryResults(i);
+
+                final List<Chunk> chunks = content.getAllChunks();
+                final int numChunks = chunks.size();
+                // We are looking only at the first and last chunks
+                for (int j = 0; j < numChunks; j+=(numChunks-1)) {
+
+                    chunkCount++;
+
+                    final Chunk c = chunks.get(j);
+                    final CloseableIterator<SAMRecord> iter = ((SamReader.PrimitiveSamReaderToSamReaderAdapter) reader).iterator(new BAMFileSpan(c));
+                    try {
+                        final SAMRecord sam = iter.next();
+                        iter.close();
+                    } catch (final Exception e) {
+                        throw new SAMException("Exception querying chunk " + j + " from reference index " + i, e);
+                    }
+                }
+
+                // also seek to first and last position in the linear index
+                final long linearIndexEntries[] = content.getLinearIndex().getIndexEntries();
+                for (int k = 0; k < linearIndexEntries.length; k+=(linearIndexEntries.length-1)) {

```

Comment:
Same suggestion here to substitute:
`for (final int binNo : Arrays.asList(0, linearIndexEntries.length - 1))`